sdk/core/azure-core/src/main/java/com/azure/core/util/paging/ContinuablePagedFluxCore.java

```diff
@@ -0,0 +1,156 @@
+    private final int defaultPrefetch;
+
+    /**
+     * Creates an instance of {@link ContinuablePagedFluxCore}.
+     *
+     * @param pageRetrieverProvider a provider that returns Page Retriever Function.
+     */
+    protected ContinuablePagedFluxCore(Supplier<Function<C, Flux<P>>> pageRetrieverProvider) {
+        this.pageRetrieverProvider = Objects.requireNonNull(pageRetrieverProvider,
+            "'pageRetrieverProvider' function cannot be null.");
+        this.defaultPrefetch = -1;
+    }
+
+    /**
+     * Creates an instance of {@link ContinuablePagedFluxCore}.
+     *
+     * @param pageRetrieverProvider a provider that returns Page Retriever Function.
+     * @param prefetch the number of Pages to be pre-fetched from the Page Retriever Function upon
+     *                 subscription.
+     */
+    protected ContinuablePagedFluxCore(Supplier<Function<C, Flux<P>>> pageRetrieverProvider, int prefetch) {
+        this.pageRetrieverProvider = Objects.requireNonNull(pageRetrieverProvider,
+            "'pageRetrieverProvider' function cannot be null.");
+        if (prefetch <= 0) {
+            throw new IllegalArgumentException("prefetch > 0 required but provided: " + prefetch);
+        }
+        this.defaultPrefetch = prefetch;
+    }
+
+    @Override
+    public Flux<P> byPage() {
+        return byPageIntern(this.pageRetrieverProvider, null, this.defaultPrefetch);
+    }
+
+    @Override
+    public Flux<P> byPage(C continuationToken) {
+        if (continuationToken == null) {
+            return Flux.empty();
+        }
+        return byPageIntern(this.pageRetrieverProvider, continuationToken, this.defaultPrefetch);
+    }
+
+    /**
+     * Flux of {@link ContinuablePage} that this Paged Flux represents.
+     *
+     * @param prefetch the number of Pages to be pre-fetched from the Page Retriever Function upon
+     *                 subscription
+     * @return a Flux of {@link ContinuablePage} that this Paged Flux represents.
+     */
+    public Flux<P> byPage(int prefetch) {
+        if (prefetch <= 0) {
+            throw new IllegalArgumentException("prefetch > 0 required but provided: " + prefetch);
+        }
+        return byPageIntern(this.pageRetrieverProvider, null, prefetch);
+    }
+
+    /**
+     * Flux of {@link ContinuablePage} identified by a continuation token.
+     *
+     * @param continuationToken the token to identify the pages to be retrieved
+     * @param prefetch the number of Pages to be pre-fetched from the Page Retriever Function upon
+     *                 subscription
+     * @return a Flux of {@link ContinuablePage} identified by the given continuation token
+     */
+    public Flux<P> byPage(C continuationToken, int prefetch) {
+        if (continuationToken == null) {
+            return Flux.empty();
+        }
+        if (prefetch <= 0) {
+            throw new IllegalArgumentException("prefetch > 0 required but provided: " + prefetch);
```

Comment:
Instead of throwing, should we be returning a Flux.error?

Reply:
right, I've updated the code.